blackout/parserapp/services/parser.py

```diff
@@ -66,51 +69,94 @@ def scrap_data(page_data):
+    else:
+        prefix = None
+        if "pages" not in os.listdir():
+            os.makedirs("pages")
+
+    with open(f"{prefix}pages/{date[1:]}.sjson", "w", encoding='utf-8') as f:
+        json.dump(data, f, ensure_ascii=False, indent=4)
+        f.close()
+    return len(data)
 
-def save_data(data):
+
+def save_data():
     connection = sqlite3.connect("../../db.sqlite3")
     cur = connection.cursor()
     summary = 0
     streets_number = 0
 
-    for row in data:
-        region = row["district"].title()
-        otg = row["otg"].title()
-        city = row["np"].title()
-        street = row["street"].title()
-        buildings = row["buildings"]
-
-        interruption = save_interruption(
-            cause=row["cause"],
-            time_off=row["time_off"],
-            time_on=row["time_on"],
-            cur=cur,
-            con=connection
-        )
-
-        exist = cur.execute(
-            f"SELECT * FROM parserapp_streets WHERE Name='{street}' AND City='{city}' AND OTG='{otg}';"
-        ).fetchall()
-
-        if len(exist) < 1:
-            cur.execute(
-                f"INSERT INTO parserapp_streets(Name, City, OTG, Region) VALUES('{street}', '{city}', '{otg}', '{region}')"
+    if "parserapp" in os.listdir():
+        path = "parserapp/services/pages"
+        files = os.listdir(path=path)
+    else:
+        path = "pages"
+        files = os.listdir(path=path)
+
+    for file in files:
+        if ".sjson" not in file:
+            files.remove(file)  # Removes all not json files
+
+    for file in files:
+        with open(f"{path}/{file}", "r") as json_data:
+            data = json.load(json_data)
+        for row in data[:10]:
+            region = row["district"].title()
+            otg = row["otg"].title()
+            city = row["np"].title()
+            street = row["street"].title()
+            buildings = row["buildings"]
+
+            interruption = save_interruption(
+                cause=row["cause"],
+                time_off=row["time_off"],
+                time_on=row["time_on"],
+                cur=cur,
+                con=connection
             )
-            connection.commit()
-            print(f"{street} added")
-            streets_number += 1
-
-            street_id = cur.lastrowid  # returns id of the last inserted record
-            buildings_number = save_buildings(buildings, street_id, cur, connection, interruption)
-            summary += buildings_number
-
-        else:
-            print(f"Street {street} already exist.")
-            street_id = exist[0][0]  # exist[0] returns record (id, Name, City, OTG, Region)
-            buildings_number = save_buildings(buildings, street_id, cur, connection, interruption)
-            summary += buildings_number
+            if street != "Не Визначена" and '"' not in street:
+                exist = cur.execute(
+                    f"SELECT * FROM parserapp_streets WHERE Name='{street}' AND City='{city}' AND OTG='{otg}';"
```

Comment:
Is it possible to rewrite using orm

Reply:
Yes, but should I rewrite it if user don't proceed any data and can't hurt DB?